src/model/chat.ts

```diff
@@ -65,15 +65,15 @@ export class ChatModel<
   }
 
   protected async runModel(
-    { handleUpdate, ...params }: Model.Chat.Run & Model.Chat.Config,
+    { handleUpdate, opts, ...params }: Model.Chat.Run & Model.Chat.Config,
     context: CustomCtx
   ): Promise<Model.Chat.Response> {
     const start = Date.now();
 
     // Use non-streaming API if no handler is provided
     if (!handleUpdate) {
       // Make the OpenAI API request
-      const response = await this.client.createChatCompletion(params);
+      const response = await this.client.createChatCompletion(params, opts);
```

Comment:
since all of the models end up calling an `openai-fetch` method that accepts `RequestOpts`, should we add `opts?: RequestOpts` to the generic model run args and pass it through for all 4 models? (chat, completion, dense, sparse).

i can see the header option (see comment below) being potentially useful when we start using non-openai endpoints that require custom headers.